receiver/sshcheckreceiver/scraper.go

```diff
@@ -0,0 +1,156 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sshcheckreceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/sshcheckreceiver"
+
+import (
+	"context"
+	"errors"
+	"runtime"
+	"time"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/sshcheckreceiver/internal/configssh"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/sshcheckreceiver/internal/metadata"
+)
+
+var errClientNotInit = errors.New("client not initialized")
+
+type sshcheckScraper struct {
+	*configssh.Client
+	*Config
+	settings component.TelemetrySettings
+	mb       *metadata.MetricsBuilder
+}
+
+// start starts the scraper by creating a new SSH Client on the scraper
+func (s *sshcheckScraper) start(_ context.Context, host component.Host) error {
+	var err error
+	s.Client, err = s.Config.ToClient(host, s.settings)
+	return err
+}
+
+func (s *sshcheckScraper) scrapeSSH(now pcommon.Timestamp) error {
+	var success int64
+
+	start := time.Now()
+	err := s.Client.Dial(s.Config.SSHClientSettings.Endpoint)
+	if err != nil {
+		s.mb.RecordSshcheckDurationDataPoint(now, time.Since(start).Milliseconds(), s.Config.SSHClientSettings.Endpoint)
+		s.mb.RecordSshcheckStatusDataPoint(now, success, s.Config.SSHClientSettings.Endpoint)
+		return err
+	}
+
+	success = 1
+	s.mb.RecordSshcheckDurationDataPoint(now, time.Since(start).Milliseconds(), s.Config.SSHClientSettings.Endpoint)
+	s.mb.RecordSshcheckStatusDataPoint(now, success, s.Config.SSHClientSettings.Endpoint)
+	return nil
```

Comment:
```suggestion
	if err == nil {
		success = 1
	}
	s.mb.RecordSshcheckDurationDataPoint(now, time.Since(start).Milliseconds(), s.Config.SSHClientSettings.Endpoint)
	s.mb.RecordSshcheckStatusDataPoint(now, success, s.Config.SSHClientSettings.Endpoint)
	return err
```